src/platform/assets/composables/useAssetFilterOptions.ts

```diff
@@ -8,12 +8,13 @@ import type { AssetItem } from '@/platform/assets/schemas/assetSchema'
  * Composable that extracts available filter options from asset data
  * Provides reactive computed properties for file formats and base models
  */
-export function useAssetFilterOptions(assets: AssetItem[] = []) {
+export function useAssetFilterOptions(assetsGetter: () => AssetItem[]) {
```

Comment:
If you really want to embrace the Vue-ism, you could use `MaybeRefOrGetter` here